### PR TITLE
Add get_status to stream

### PR DESCRIPTION
**Pull request by britasys**
This commit addresses the requirements outlined in issue #228, enhancing the module's functionality and supporting better status tracking.

This is my first commit to this repository. So, I need help.
I didn't know where to put the structure.

@ashtum

### CODE REVIEW

**Comment by vinniefalco** [comment]
This needs a test, and to ensure all new lines have coverage.

**Comment by britasys** [comment]
I have added a test for the get_status function and made some modifications to the initial commit.

File: include/boost/beast/websocket/stream.hpp [hunk 1]
```diff
@@ -420,7 +413,7 @@ class stream
 
     /// Get the permessage-deflate status
     void
-    get_status(permessage_deflate_status& status);
+    get_status(detail::permessage_deflate_status &status);
```
Comment (vinniefalco) [inline include/boost/beast/websocket/stream.hpp:416]: `detail` symbols are not supposed to be in public APIs

**Comment by ashtum** [comment]
@britasys, please move the `struct permessage_deflate_status` to the top of `stream.hpp` and add Javadoc comments for the struct and its members. Initialize the members with default values to minimize the risk of undefined behavior in users' code. For reference, see how it's done for the `permessage_deflate` struct: https://github.com/boostorg/beast/blob/fee9be0be10c9c9a22ac1505a710d1d8ed5a3dfb/include/boost/beast/websocket/option.hpp#L32

File: include/boost/beast/websocket/stream.hpp (hunk 1 above)
Comment (britasys) [reply include/boost/beast/websocket/stream.hpp]: The 'permessage_deflate_status' structure has been relocated to 'stream.hpp' as @ashtum suggested.

File: include/boost/beast/websocket/impl/stream.hpp [hunk 2]
```diff
@@ -175,6 +175,18 @@ get_option(timeout& opt)
     opt = impl_->timeout_opt;
 }
 
+template <class NextLayer, bool deflateSupported>
+void
+stream<NextLayer, deflateSupported>::
+get_status(permessage_deflate_status &status)
```
Comment (ashtum) [inline include/boost/beast/websocket/impl/stream.hpp:181]: `get_status` should be marked `const noexcept`.

File: include/boost/beast/websocket/detail/impl_base.hpp [hunk 3]
```diff
@@ -345,6 +345,12 @@ struct impl_base<true>
             result = clamp(result, rd_msg_max);
         return result;
     }
+
+    void
+    get_pmd_config(detail::pmd_offer &pmd)
```
Comment (ashtum) [inline include/boost/beast/websocket/detail/impl_base.hpp:350]: Please rename this to `get_config_pmd` to be consistent with `get_option_pmd`.

File: include/boost/beast/websocket/detail/impl_base.hpp [hunk 4]
```diff
@@ -495,6 +501,16 @@ struct impl_base<false>
...
+    {
+        pmd.accept = false;
+        pmd.server_max_window_bits = 0;
+        pmd.client_max_window_bits = 0;
+        pmd.server_no_context_takeover = false;
+        pmd.client_no_context_takeover = false;
```
Comment (ashtum) [inline include/boost/beast/websocket/detail/impl_base.hpp:512]: You might want to change these to `pmd = {};` instead.

File: include/boost/beast/websocket/stream.hpp [hunk 5]
```diff
@@ -38,6 +38,30 @@
...
 namespace websocket {
+/** permessage-deflate extension status.
+
+    These settings indicate the status of the permessage-deflate
+    extension, showing if it is active and the window bits in use.
+
+    @note
+
+    Used to check the status of the permessage-deflate extension after
+    the WebSocket handshake.
```
Comment (ashtum) [inline include/boost/beast/websocket/stream.hpp:49]: Please move this note to the Javadoc of `stream::get_status`. Please squash all the commits in a single commit with a message like 'Make permessage-deflate status queryable in websocket::stream'.

**Review by ashtum — APPROVED** [review]

**Comment by britasys** [comment]
Hi, Do I have to do anything else?

**Comment by ashtum** [comment]
> Hi, Do I have to do anything else?

No, everything looks good to me. I've delayed the merge until the Boost 1.86 release, just in case we need to push a last-minute hotfix to develop branch. This will be included in Beast in Boost 1.87. Thank you.

**Pull request merged**